### PR TITLE
For -> forEach + use path module

### PR DESCRIPTION
**Pull request by xasos**
- Switched from using RegEx to using Node's standard path library to get the extension name
- Switched for loop to for each on files array.

### CODE REVIEW

File: lslib.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+        if(path.extname(entry) == ext) {
+          result.push(entry);
         }
-      }
+      });
       callback(null, result);
     }
   });
-};
+};
```
Comment (frasertweedale) [inline lslib.js]: Please retain the newline at end of file.

File: lslib.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
     if (err) {
       callback(err, null);
     }
     else {
       result = [];
-      for (i = 0; i < files.length; i++) {
-        if (pat.test(files[i])) {
-          result.push(files[i]);
+      files.forEach(function(entry) {
+        if(path.extname(entry) == ext) {
```
Comment (frasertweedale) [inline lslib.js]: [nitpick] please include a space after the `if`

**Comment by frasertweedale** [comment]
The "Make it Modular" test fails with this change.  At a glance, it could be that `path.extname`
includes the `.` in its result, and you may be comparing that to the file extension _without_ the period.

**Comment by frasertweedale** [comment]
Thanks for the PR.  Overall I am very much in favour of this change!

**Comment by xasos** [comment]
Thanks, will change those right now!<br> Just curious, which test did it fail specifically? I've been getting the correct output and the correct # of elements in the last array, but it seems that the last check is failing [for a couple of people](https://github.com/nodeschool/discussions/issues/707#issuecomment-67784500) and nodeschool is aware of it.

**Comment by frasertweedale** [comment]
Test output (sorry, GitHub is butchering the formatting; maybe it doesn't parse email comments as markdown)

```
    [bacardi:~/dev/learnyounode-solutions] [ 6a1e2c4... ] fraser% node node_modules/.bin/learnyounode verify ls_modular.js

    Your submission results compared to the expected:

                     ACTUAL                                  EXPECTED
    ────────────────────────────────────────────────────────────────────────────────

       ""                                  !=    "CHANGELOG.md"
                                           !=    "LICENCE.md"
                                           !=    "README.md"
                                           !=    ""

    ────────────────────────────────────────────────────────────────────────────────

    ✗ Submission results match expected
    ✓ Additional module file exports a single function
    ✓ Additional module file exports a function that takes 3 arguments
    ✓ Additional module file handles errors properly
    ✓ Additional module file handles callback argument
    ✓ Additional module file returned two arguments on the callback function
    ✓ Additional module file returned Array as second argument of the callback
    ✗ Your additional module file [lslib.js] did not return an Array with the correct number of elements as the second argument of the callback

    # FAIL

    Your solution to MAKE IT MODULAR didn't pass. Try again!
```

**Comment by xasos** [comment]
The error was a lack of a period to the extension, like you mentioned above. It has since been updated in latest commit and now passes 100%:

```
✓ Submission results match expected
✓ Additional module file exports a single function
✓ Additional module file exports a function that takes 3 arguments
✓ Additional module file handles errors properly
✓ Additional module file handles callback argument
✓ Additional module file returned two arguments on the callback function
✓ Additional module file returned Array as second argument of the callback
✓ Additional module file returned correct number of elements as the second argument of the callback
✓ Additional module file returned correct list of files as the second argument of the callback

# PASS
```

File: lslib.js [hunk 3]
```diff
@@ -1,19 +1,21 @@
 var fs = require('fs');
+var path = require('path');
 
 module.exports = function(dirname, ext, callback) {
-  var pat = RegExp('\\.' + ext + '$');
+  extension = "." + ext;
```
Comment (frasertweedale) [inline lslib.js]: Needs 'var' for lexical scope.  (Alternatively, move the append down to where the comparison happens,
i.e. `if (path.extname(entry) == "." + ext) {`

File: lslib.js [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
+      files.forEach(function(entry) {
+        if (path.extname(entry) == extension) {
+          result.push(entry);
         }
-      }
+      });
       callback(null, result);
     }
   });
+
```
Comment (frasertweedale) [inline lslib.js:20]: extraneous newline

**Comment by frasertweedale** [comment]
So close now.  Couple other comments to address, and could you please squash your commits down into a single commit?

**Comment by xasos** [comment]
Whoops didn't catch the `var`, everything should be updated now :)

**Comment by frasertweedale** [comment]
Merged; thanks!

**Pull request merged**